includes/class-omise-money.php

```diff
@@ -0,0 +1,87 @@
+<?php
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * @since 3.5
+ */
+class Omise_Money {
+	/**
+	 * @var float
+	 */
+	protected $amount;
+
+	/**
+	 * @var string
+	 */
+	protected $currency;
+
+	/**
+	 * @var array
+	 */
+	private $subunit_multiplier = array(
+		'AUD' => 100,
+		'CAD' => 100,
+		'CHF' => 100,
+		'CNY' => 100,
+		'DKK' => 100,
+		'EUR' => 100,
+		'GBP' => 100,
+		'HKD' => 100,
+		'JPY' => 1,
+		'MYR' => 100,
+		'SGD' => 100,
+		'THB' => 100,
+		'USD' => 100
+	);
+
+	/**
+	 * @param int|float|string $amount
+	 * @param string           $currency
+	 */
+	public function __construct( $amount, $currency ) {
+		if ( ! isset( $this->subunit_multiplier[ strtoupper( $currency ) ] ) ) {
+			throw new Exception( __( 'We do not support the currency you are using.', 'omise' ) );
+		}
+
+		$this->amount   = (float) $this->purifyAmount( $amount );
+		$this->currency = strtoupper( $currency );
```

Comment:
Note, the part that I am still 50/50 is that, should we transform a currency string that is passed at the constructor to an upper case or better just leave it as it is (whatever, lower-case or upper-case).

All opinions are welcomed.